src/fusor/fusion_matching.py

```diff
@@ -0,0 +1,278 @@
+            else:
+                return False
+
+        # Compare other structural elements
+        match_data_5prime = self._compare_structure(
+            assayed_transcript_segments[0], categorical_transcript_segments[0], True
+        )
+        if not match_data_5prime[0]:
+            return False
+        match_data_3prime = self._compare_structure(
+            assayed_transcript_segments[1], categorical_transcript_segments[1], False
+        )
+        if not match_data_3prime[0]:
+            return False
+        return True, match_score + match_data_5prime[1] + match_data_3prime[1]
+
+    async def match_fusion(
+        self,
+        assayed_fusion: AssayedFusion,
+        sources: tuple[FusionSources],
+        cache_dir: Path,
+    ) -> list[tuple[CategoricalFusion, int]] | None:
+        """Return best matching fusion
+        :param assayed_fusion: The assayed fusion object
+        :param sources: A list of FusionSources
+        :param cache_dir: The location where the cached categorical fusion objects are stored
+        :return A list of tuples containing matching categorical fusion objects and their associated match score or None
+        """
+        matched_fusions = []
+        categorical_fusion_classes = {FusionSources.CIVIC: CIVICCategoricalFusions}
+        sources = {k: v for k, v in categorical_fusion_classes.items() if k in sources}
+        for source in sources:
+            source_class = sources[source]
+            source_class = source_class(self.translator)
+            cache_path = cache_dir / source_class.cache_name
+            if not cache_path.exists():
+                await source_class.save_categorical_fusions(output_dir=cache_path)
+            self.categorical_fusions.extend(
+                self.load_categorical_fusions(output_dir=cache_path)
+            )
+        categorical_fusions = self._filter_categorical_fusions(
+            assayed_fusion, self.categorical_fusions
+        )
+        for categorical_fusion in categorical_fusions:
+            match_information = self._compare_fusion(assayed_fusion, categorical_fusion)
+            if match_information:
+                matched_fusions.append((categorical_fusion, match_information[1]))
+        return (
+            sorted(matched_fusions, key=lambda x: x[1], reverse=True)
+            if matched_fusions
+            else None
+        )
+
+
+class CIVICCategoricalFusions(FusionMatcher):
+    """Class for loading and generating cache of CIViC Categorical Fusions"""
+
+    cache_name = "civic_translated_fusions.pkl"
+
+    async def save_categorical_fusions(self, output_dir: Path) -> None:
+        """Load fusion variants from CIViC and convert to CategoricalFusion objects
+
+        :param output_dir: The location where the cache will be stored
+        """
+        # Load in all accepted fusions variants
+        variants = civic.get_all_fusion_variants(include_status="accepted")
```

Comment:
@korikuzma Is it ok to leave this line here? Not sure if the fusions from CIViC need to be accessed in a different location